bach/bach/dataframe.py

```diff
@@ -2760,6 +2760,40 @@ def _get_parsed_subset_of_data_columns(
             )
         return subset
 
+    def stack(self, dropna: bool = True) -> 'Series':
+        """
+        Stacks all data_columns into a single index series.
+
+        :param dropna: Whether to drop rows that contain missing values. If the caller has
+            at least an index series, this might generate different combinations between
+            the index and the stacked values.
+
+        :return: a reshaped series that includes a new index (named "__stacked_index")
+            containing the caller's column labels as values.
+        .. note::
+            ``level`` parameter is not supported since multilevel columns are not allowed.
+        """
+        df = self.copy()
+        if df.group_by:
+            df = df.materialize()
+
+        dc_dfs = []
+        # convert each data column series to DataFrame and use series name as new index value
+        for series_name, series in df._data.items():
+            dc_df = df.all_series[series_name].copy_override(name='__stacked').to_frame()
```

Comment:
```suggestion
        for series_name, series in df.data.items():
            dc_df = series.copy_override(name='__stacked').to_frame()
```

minor thing. `series` was never used.

Reply:
afff thanks